foundry/foundry.py

```diff
@@ -857,6 +857,53 @@ def toTorch(self, raw=None, split=None):
             return FoundryDataset_Torch(inputs, targets)
         else:
             raise NotImplementedError
+
+    def toTensorflow(self, raw=None, split=None):
+        """Convert Foundry Dataset to a Tensorflow Sequence
+
+        Arguments:
+            raw (dict): The output of running ``f.load_data(as_hdf5=False)``
+                    Recommended that this is left as ``None``
+                    **Default:** ``None``
+            split (string): Split to create Tensorflow Sequence on.
+                    **Default:** ``None``
+
+        Returns: (FoundryDataset_TF) Tensorflow Sequence of all the data from the specified split
+
+        """
+        if not raw:
+            raw = self.load_data(as_hdf5=False)
+        
+        if not split:
+            split = self.dataset.splits[0].type
+
+        if self.dataset.data_type.value == "hdf5":
+            inputs = []
+            targets = []
+            for key in self.dataset.keys:
+                if len(raw[split][key.type][key.key[0]].keys()) != self.dataset.n_items:
```

Comment:
Complex code (e.g., a 4-nested dictionary access) is going to be hard to maintain later. Make sure to add in some line comments (I like one per ~5 lines of code, fewer if the code is complicated)

Reply:
This was marked as resolved so I didn't check it... oops.